src/main/kotlin/uk/gov/justice/digital/hmpps/crimeportalgateway/xml/DocumentUtils.kt

```diff
@@ -0,0 +1,44 @@
+package uk.gov.justice.digital.hmpps.crimeportalgateway.xml
+
+import org.w3c.dom.Element
+import org.w3c.dom.Node
+import org.w3c.dom.NodeList
+import javax.xml.xpath.XPath
+import javax.xml.xpath.XPathConstants
+import javax.xml.xpath.XPathFactory
+
+object DocumentUtils {
+
+    private const val OU_CODE_LENGTH = 5
+
+    private const val SOURCE_FILE_NAME_EXPR = "//source_file_name/text()"
+
+    private const val DELIMITER = "_"
+
+    fun getCourtCode(documents: Element): String? {
+        // XPathFactory not thread safe so make one each time
+        val xPath: XPath = XPathFactory.newInstance().newXPath()
+        val exp = xPath.compile(SOURCE_FILE_NAME_EXPR)
+        val nodeList = exp.evaluate(documents, XPathConstants.NODESET) as NodeList
+        for (i in 0 until nodeList.length) {
+            getOuCode(nodeList.item(i))?.let {
+                code ->
```

Comment:
The dps-gradle-spring-boot seems to require this strange format in its linting checks